### PR TITLE
Improve software `exportAnimatedCamera`

### PR DESCRIPTION
**Pull request by gregoire-dl**
## Features list
- [x] Remove outdated `convertAnimatedCamera`
- [x] Add image path regex filter option in `exportAnimatedCamera`
- [x] Add dslr / video camera export methods in `exportAnimatedCamera`
- [x] Add `startFrame` parameter in function `initAnimatedCamera` of `AlembicExporter`

### CODE REVIEW

File: src/software/export/main_exportAnimatedCamera.cpp [hunk 1]
```diff
@@ -89,86 +108,174 @@ int main(int argc, char** argv)
...
+       !std::regex_match(view.getImagePath(), regexFilter))
       continue;
 
-    const unsigned int frame = std::stoul(imagePath.substr(lastCharIndex + 1));
+    // pose and intrinsic defined
+    if(!sfmData.isPoseAndIntrinsicDefined(&view))
+      continue;
+
+    const std::string imagePathStem = fs::path(viewPair.second->getImagePath()).stem().string();
+    std::size_t lastCharIndex = imagePathStem.find_last_not_of("0123456789");
```
Comment (fabiencastan) [inline src/software/export/main_exportAnimatedCamera.cpp]: A regex would be more straightforward.

File: src/software/export/main_exportAnimatedCamera.cpp [hunk 2]
```diff
@@ -89,86 +108,174 @@ int main(int argc, char** argv)
...
+  std::regex regexFilter;
 
-  sfm::SfMData sfmDataFilter;
-  if(!sfm::Load(sfmDataFilter, sfmDataFilterFilename, sfm::ESfMData::VIEWS))
-  {
-    ALICEVISION_LOG_ERROR("The input SfMData file '" << sfmDataFilterFilename << "' cannot be read.");
-    return EXIT_FAILURE;
-  }
+  if(!viewFilter.empty())
+    regexFilter = std::regex(viewFilter);
```
Comment (fabiencastan) [inline src/software/export/main_exportAnimatedCamera.cpp]: You can convert the user filter into a regex:

```
filterToRegex = std::regex_replace( filterToRegex, std::regex( "\\*" ), "(.*)" );
filterToRegex = std::regex_replace( filterToRegex, std::regex( "\\?" ), "(.)" );
filterToRegex = std::regex_replace( filterToRegex, std::regex( "\\@" ), "[0-9]+" ); // one @ correspond to one or more digits
filterToRegex = std::regex_replace( filterToRegex, std::regex( "\\#" ), "[0-9]" ); // each # in pattern correspond to a digit
```

File: src/software/export/main_exportAnimatedCamera.cpp (hunk 1 above)
Comment (gregoire-dl) [reply src/software/export/main_exportAnimatedCamera.cpp]: done

File: src/software/export/main_exportAnimatedCamera.cpp (hunk 2 above)
Comment (gregoire-dl) [reply src/software/export/main_exportAnimatedCamera.cpp]: done

**Pull request merged**